platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/GenesisStateBuilder.java

```diff
@@ -34,31 +34,37 @@ public final class GenesisStateBuilder {
     private GenesisStateBuilder() {}
 
     /**
-     * Construct a genesis platform state.
+     * Initializes a genesis platform state.
      *
-     * @return a genesis platform state
      */
-    private static PlatformState buildGenesisPlatformState(
-            final AddressBook addressBook, final SoftwareVersion appVersion) {
+    private static void initGenesisPlatformState(
+            final PlatformContext platformContext,
+            final PlatformStateAccessor platformState,
+            final AddressBook addressBook,
+            final SoftwareVersion appVersion) {
+        platformState.bulkUpdate(v -> {
+            v.setAddressBook(addressBook.copy());
+            v.setCreationSoftwareVersion(appVersion);
+            v.setRound(0);
+            v.setLegacyRunningEventHash(null);
+            v.setConsensusTimestamp(Instant.ofEpochSecond(0L));
 
-        final PlatformState platformState = new PlatformState();
-        platformState.setAddressBook(addressBook.copy());
+            final BasicConfig basicConfig = platformContext.getConfiguration().getConfigData(BasicConfig.class);
 
-        platformState.setCreationSoftwareVersion(appVersion);
-        platformState.setRound(0);
-        platformState.setLegacyRunningEventHash(null);
-        platformState.setConsensusTimestamp(Instant.ofEpochSecond(0L));
-
-        return platformState;
+            final long genesisFreezeTime = basicConfig.genesisFreezeTime();
+            if (genesisFreezeTime > 0) {
+                platformState.setFreezeTime(Instant.ofEpochSecond(genesisFreezeTime));
+            }
```

Comment:
shouldn't this be operating on `v`, rather than `platformState`?

Reply:
Good catch. Yes, it should